web/src/components/tasks/search-business-name/SearchBusinessNameForm.tsx

```diff
@@ -51,40 +48,104 @@ export const SearchBusinessNameForm = (props: Props): ReactElement => {
 
   const { Config } = useConfig();
   const didInitialSearch = useRef<boolean>(false);
-  const { userData } = useUserData();
-  const [nameAvailability, setNameAvailability] = useState<NameAvailability | undefined>(undefined);
-  const [submittedName, setSubmittedName] = useState<string>("");
+  const { userData, updateQueue } = useUserData();
+  const {
+    setFormationFormData,
+    setBusinessNameAvailability,
+    setDbaBusinessNameAvailability,
+    setFieldsInteracted,
+  } = useContext(BusinessFormationContext);
+
+  const setNameAvailability = props.isDba ? setDbaBusinessNameAvailability : setBusinessNameAvailability;
 
+  const FIELD_NAME = "businessName";
   const SearchBusinessNameErrorLookup: Record<SearchBusinessNameError, string> = {
     BAD_INPUT: Config.searchBusinessNameTask.errorTextBadInput,
     SEARCH_FAILED: Config.searchBusinessNameTask.errorTextSearchFailed,
   };
 
-  useEffect(() => {
-    props.onChange && props.onChange(nameAvailability);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [nameAvailability]);
+  const emptyNameAvailability: NameAvailability = {
+    similarNames: [],
+    status: undefined,
+    lastUpdatedTimeStamp: "",
+  };
 
   const Unavailable = props.unavailable;
   const Available = props.available;
-  const onSubmit = props.onSubmit;
 
   const doSearch = useCallback(
     async (
       event: FormEvent<HTMLFormElement> | undefined,
       { isInitialSubmit }: { isInitialSubmit: boolean }
     ): Promise<void> => {
       searchBusinessName(event)
-        .then(({ nameAvailability, submittedName }) => {
-          setNameAvailability(nameAvailability);
-          setSubmittedName(submittedName);
-          if (onSubmit) {
-            onSubmit(submittedName, nameAvailability, isInitialSubmit);
+        .then(async ({ nameAvailability, submittedName }) => {
+          if (props.isDba) {
+            if (!nameAvailability || !updateQueue) {
+              return;
+            }
+            setNameAvailability(nameAvailability);
+            if (nameAvailability.status === "AVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                  },
+                  dbaBusinessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  nexusDbaName: submittedName,
```

Comment:
Why do we have this data structure where `businessName` exists in both `profileData` and `formationFormData` indepedently, and have to keep each in line with each other, whereas for `nexusDbaName` we don't store it in `formationFormData` at all and *only* update profile data when something is entered/searched here?

Reply:
I honestly couldn't tell you, and I think that should be looked at as a follow on to this.

Reply:
Based on our types it looks like `businessName` is also in `TaxFilingData`. I'm not looking super deep into the code, but there may be potentially 3 places we store the business name.

Reply:
[Refactor how businessName is stored](https://www.pivotaltracker.com/story/show/185356746)